test_pull_requests

```diff
@@ -152,6 +160,40 @@ def submitted_tests_for_branch(branch)
   $submitted_tests[branch] ? $submitted_tests[branch] : $submitted_tests['*']
 end
 
+def extract_bot_comment_metadata(bot_comment_body, settings)
+  if bot_comment_body =~ /^#{settings['test_prefix']} ([^(]+) \(/
+    content = $1 || ''
+  else
+    raise "'#{settings['name']}' pull request not found"
+  end
+  build_url = bot_comment_body[/ (\(https:[a-f0-9]{40}\))/, 1] || ''
+  base_commit = bot_comment_body[/ (\(Base Commit: [a-f0-9]{40}\))/, 1] || ''
+  extended_tests = bot_comment_body[/ (\(Extended Tests: .+)/, 1] || ''
+  return content, build_url, base_commit, extended_tests
+end
+
+def compose_bot_comment(prefix, comment, metadata={})
+  updated_comment = "#{prefix} #{comment}"
+  for metadata_tag in METADATA_TAG_NAMES.keys
+    if metadata[metadata_tag] && (! metadata[metadata_tag].empty?)
+      updated_comment += if metadata[metadata_tag].start_with? '('
+                           " #{metadata[metadata_tag]}"
+                         else
+                           if METADATA_TAG_NAMES[metadata_tag] == :no_prefix
```

Comment:
You might as well use each do |key, value| if you are going to then lookup values.